jetty-ee10/jetty-ee10-servlet/src/main/java/org/eclipse/jetty/ee10/servlet/ServletChannelState.java

```diff
@@ -559,19 +576,25 @@ public void errorHandling()
         }
     }
 
-    public void errorHandlingComplete()
+    public void errorHandlingComplete(Throwable failure)
     {
         boolean handle;
         try (AutoLock ignored = lock())
         {
             if (LOG.isDebugEnabled())
-                LOG.debug("errorHandlingComplete {}", toStringLocked());
+                LOG.debug("errorHandlingComplete {}", toStringLocked(), failure);
 
             handle = _state == State.WAITING;
             if (handle)
                 _state = State.WOKEN;
+
+            // If there is a failure while trying to
+            // handle a previous failure, just bail out.
+            if (failure != null)
+                abortResponse(failure);
+
             if (_requestState == RequestState.ERRORING)
-                _requestState = RequestState.COMPLETE;
+                _requestState = failure == null ? RequestState.COMPLETE : RequestState.COMPLETED;
```

Comment:
Need a comment to explain this, as I can't get my head around it (even though I think I suggested it).

If the requestState is still in ERRORING, then it means the ErrorHandler did not change it's state, so therefore `startAsync` or `sendError` have not been called. 

is `failure == null` the right test?  should it not be testing is we are in an async cycle?   Also how can failure be non-null give the test directly above?  something is not right here?

Reply:
I changed it to just COMPLETE.

The reason for the check was that if we failed hard trying to write an error page, there is no point going to COMPLETE and trying to flush the `HttpOutput`, so we just go to COMPLETED.

However, there should be no hard trying to flush `HttpOutput` as it can fail in other circumstances and its failure should be handled properly by going to COMPLETED and then terminating.